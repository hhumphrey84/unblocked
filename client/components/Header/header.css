@import '../../assets/colors.css';
@import '../../assets/dimensions.css';
@import '../../assets/fonts.css';

:root {
    --logo-dimension: 24px;
}

.wrapper {
    padding: calc(var(--spacing) * 2);
    margin-bottom: calc(var(--spacing) * 2);
    width: 100%;
}

.branding {
    display: flex;
    flex-direction: row;
}

.logo {
    width: var(--logo-dimension);
    height: var(--logo-dimension);
    margin-right: var(--spacing);
}

.title {
    color: var(--color-black);
    font-family: var(--font-family-2);
    font-size: 18px;
    line-height: var(--logo-dimension);
    font-weight: var(--weight-bold);
}
