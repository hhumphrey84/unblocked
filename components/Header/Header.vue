<template>
    <header
        :data-test-id="testIds.WRAPPER"
        class="wrapper"
    >
        <div class="branding">
            <span class="logo">
                <nuxt-link
                    :data-test-id="testIds.HOME_LINK"
                    to="/"
                >
                    <icon
                        id="headerLogo"
                        title="UnBlocked â€“ Back to home"
                    >
                        <logo />
                    </icon>
                </nuxt-link>
            </span>

            <h1 class="title">UnBlocked</h1>
        </div>
    </header>
</template>

<style scoped src="./header.css"></style>
<script>
    import Icon from '~/components/Icon/Icon.vue';
    import Logo from '~/components/svgs/Logo.vue';
    import testIds from './testIds.constants';

    const data = {
        testIds,
    };

    const components = {
        Logo,
        Icon,
    };

    export default {
        components,
        data: () => data,
    }
</script>
