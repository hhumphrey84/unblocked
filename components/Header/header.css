@import '../../assets/colors.css';
@import '../../assets/dimensions.css';
@import '../../assets/fonts.css';

.wrapper {
    padding: calc(var(--spacing) * 2);
    margin-bottom: calc(var(--spacing) * 2);
    width: 100%;
}

.logo {
    width: 24px;
    height: 24px;
    float: left;
    margin-right: var(--spacing);
}

.title {
    font-family: var(--font-family-2);
    font-size: 24px;
    line-height: 1em;
    font-weight: var(--weight-bold);;
    color: var(--color-black);
    float: left;
}
