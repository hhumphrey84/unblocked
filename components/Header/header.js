import Icon from '~/components/Icon/Icon.vue'
import Logo from '~/components/svgs/Logo.vue'

export default {
    components: {
        Logo,
        Icon,
    }
}
