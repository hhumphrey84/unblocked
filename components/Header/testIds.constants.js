export default {
    WRAPPER: 'header',
    HOME_LINK: 'header__homeLink',
};
