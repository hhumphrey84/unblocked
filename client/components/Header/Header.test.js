import { shallowMount } from '@vue/test-utils';

import Header from './Header.vue';

const wrapper = shallowMount(Header, { stubs: ['nuxt-link'] });

describe('Header', () => {
    test('renders properly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
})
