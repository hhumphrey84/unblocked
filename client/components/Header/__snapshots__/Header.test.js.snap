// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
<header data-test-id="header" class="wrapper">
  <div class="branding"><span class="logo"><nuxt-link data-test-id="header__homeLink" to="/"><icon-stub datatestid="icon" title="UnBlocked â€“ Back to home" id="headerLogo"><logo-stub></logo-stub></icon-stub></nuxt-link></span>
    <h1 class="title">UnBlocked</h1>
  </div>
</header>
`;
